frontend/src/containers/RecyclingBinPage/RecyclingBinPage.js

```diff
@@ -1,57 +1,311 @@
+import CustomSnackbar from '../../components/CustomSnackbar/CustomSnackbar';
 import { getUserVideosSize } from '../../api/UserAPI';
+import RestoreIcon from '../../assets/generic/RestoreIcon';
+import { normalizeCards, sortCardDates, transformCards } from '../../util/card';
+import VideoCardPage from '../../components/VideoCardPage/VideoCardPage';
+import EmptyRecyclingBinContent from '../../components/EmptyRecyclingBinContent/EmptyRecyclingBinContent';
 
 class RecyclingBinPage extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      showNavDrawer: false,
+      videosInformation: {},
       size: 0,
+      sortedVideoDates: [],
+      sortAscending: true,
+      selectedCardIds: [],
+      showUploadModal: false,
+      showNavDrawer: false,
+      showDeletionDialog: false,
+      showDeletionError: false,
+      showDeletionSuccess: false,
+      showVideoRetrievalError: false,
+      showVideoRestorationError: false,
+      showVideoRestorationSuccess: false,
+      isLoading: true,
     };
   }
 
   componentDidMount() {
+    const { sortAscending } = this.state;
+    getRecycledVideos()
+      .then((response) => {
+        if (response.data.length !== 0) {
+          const { transformedVideosInformation } = transformCards(
+            response.data,
+            'deleteDate',
+          );
+          const sortedDates = sortCardDates(
+            transformedVideosInformation,
+            sortAscending,
+          );
+          this.setState({
+            videosInformation: transformedVideosInformation,
+            sortedVideoDates: sortedDates,
+          });
+        } else {
+          this.setState({
+            videosInformation: [],
+          });
+        }
+      })
+      .catch(() => {
+        this.setState({ showVideoRetrievalError: true });
+      })
+      .finally(() => {
+        this.setState({ isLoading: false });
+      });
+
     getUserVideosSize()
       .then((response) => this.setState({ size: response.data }))
       .catch(() => {
         this.setState({ size: 0 });
       });
   }
 
+  toggleSort = () => {
+    const { sortAscending, videosInformation } = this.state;
+    this.setState({
+      sortAscending: !sortAscending,
+      sortedVideoDates: sortCardDates(videosInformation, !sortAscending),
+    });
+  };
+
+  toggleSelectAll = () => {
+    const { videosInformation } = this.state;
+    if (Object.values(videosInformation).length === 0) return;
+    const allCards = Object.values(videosInformation)
+      .reduce((acc, val) => acc.concat(val), [])
+      .map((val) => val.id);
+    this.setState({ selectedCardIds: allCards });
+  };
+
   toggleNavDrawer = () => {
     const { showNavDrawer } = this.state;
     this.setState({ showNavDrawer: !showNavDrawer });
   };
 
+  toggleDeletionDialog = () => {
+    const { showDeletionDialog } = this.state;
+    this.setState({ showDeletionDialog: !showDeletionDialog });
+  };
+
+  handleVideoDeletion = () => {
```

Comment:
There is a bug in this page and in `LibraryPage`. If you delete a video, then go to `ProfilePage` and go back to previous page, then the deleted video keeps appearing again

Reply:
Did you solve this?

Reply:
No